node/src/node.rs

```diff
@@ -137,28 +137,73 @@ where
         Ok(self.p2p.set_peer_trust(peer_id, is_trusted).await?)
     }
 
+    /// Request the head header from the network.
+    pub async fn request_head_header(&self) -> Result<ExtendedHeader> {
+        Ok(self.p2p.get_head_header().await?)
+    }
+
+    /// Request a header for the block with a given hash from the network.
+    pub async fn request_header_by_hash(&self, hash: &Hash) -> Result<ExtendedHeader> {
+        Ok(self.p2p.get_header(*hash).await?)
+    }
+
+    /// Request a header for the block with a given height from the network.
+    pub async fn request_header_by_height(&self, hash: u64) -> Result<ExtendedHeader> {
+        Ok(self.p2p.get_header_by_height(hash).await?)
+    }
+
+    /// Request headers in range (from, from + amount] from the network.
+    ///
+    /// The headers will be verified with the `from` header.
+    pub async fn request_verified_headers_range(
```

Comment:
After our discussion, I feel `_range` shouldn't be in the name. `request_verified_headers` feels nicer

Reply:
done